3-UsingAPIs/Week2/assignment/ex1-programmerFun/index.js

```diff
@@ -16,29 +16,44 @@ Full description at: https://github.com/HackYourFuture/Assignments/blob/main/3-U
    url with `.shx`. There is no server at the modified url, therefore this 
    should result in a network (DNS) error.
 ------------------------------------------------------------------------------*/
-function requestData(url) {
-  // TODO return a promise using `fetch()`
+const body = document.querySelector("body");
+const renderImg = document.createElement('img');
+
+
+async function requestData(url) {
+ try {
+  const response = await fetch(url)
+  if (!response.ok) {
+    throw new Error("SOME ERROR WITH API" +  response.status);
+  }
+  
+  return response.json()
```

Comment:
If you use a try/catch block here (but in this case no need, see below) you should also use `await` here. Otherwise, if the `response.json()` returns a rejected promise it will not be caught by the `catch` clause inside this function: it is the `await` keyword that causes an error to be thrown when it gets a rejected promise.

```suggestion
  return await response.json()
```